sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,272 @@
+    return this.casInvoiceRepo.find({
+      select: {
+        id: true,
+        casSupplier: {
+          id: true,
+          supplierNumber: true,
+          supplierAddress: true as unknown,
+        },
+        createdAt: true,
+        invoiceNumber: true,
+        disbursementReceipt: {
+          id: true,
+          fileDate: true,
+          createdAt: true,
+        },
+        casInvoiceBatch: {
+          id: true,
+          batchName: true,
+          batchDate: true,
+          createdAt: true,
+        },
+        casInvoiceDetails: {
+          id: true,
+          valueAmount: true,
+          casDistributionAccount: {
+            id: true,
+            operationCode: true,
+            awardValueCode: true,
+            distributionAccount: true,
+          },
+        },
+      },
+      relations: {
+        casInvoiceBatch: true,
+        disbursementReceipt: true,
+        casSupplier: true,
+        casInvoiceDetails: {
+          casDistributionAccount: true,
+        },
+      },
+      where: {
+        casInvoiceBatch: {
+          approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+        },
+        invoiceStatus: CASInvoiceStatus.Pending,
+      },
+      order: {
+        createdAt: "ASC",
+        casInvoiceDetails: {
+          casDistributionAccount: {
+            awardValueCode: "ASC",
+            operationCode: "ASC",
+          },
+        },
+      },
+      take: pollingRecordsLimit,
+    });
+  }
+
+  /**
+   * Send pending invoices to CAS.
+   * @param pollingRecordsLimit maximum number of records to be returned.
+   * @param processSummary parent process summary for logging.
+   */
+  async sendInvoices(
+    pollingRecordsLimit: number,
+    processSummary: ProcessSummary,
+  ): Promise<void> {
+    processSummary.info("Checking for pending invoices.");
+    const pendingInvoices = await this.getPendingInvoices(pollingRecordsLimit);
+    if (pendingInvoices.length === 0) {
+      processSummary.info("No pending invoices found.");
+    } else {
+      processSummary.info(
+        `Found ${pendingInvoices.length} pending invoice(s) sent to CAS.`,
+      );
+      // Process each invoice in parallel.
+      await processInParallel(
+        (pendingInvoice) => this.processInvoice(pendingInvoice, processSummary),
+        pendingInvoices,
+      );
+    }
+  }
+
+  /**
+   * Process and send an invoice to CAS.
+   * @param pendingInvoice pending invoice.
+   * @param parentProcessSummary parent process summary for logging.
+   */
+  private async processInvoice(
+    pendingInvoice: CASInvoice,
+    parentProcessSummary: ProcessSummary,
+  ): Promise<void> {
+    const summary = new ProcessSummary();
+    parentProcessSummary.children(summary);
+    // Log information about the pending invoice being processed.
+    summary.info(
+      `Processing pending invoice: ${pendingInvoice.invoiceNumber}.`,
+    );
+    const pendingInvoicePayload = this.getPendingInvoicePayload(pendingInvoice);
```

Comment:
This can be moved inside the `try` to allow the execution to proceed if there is a data error.